src/popup/components/TransactionList.vue

```diff
@@ -50,179 +40,230 @@ import {
     InfiniteScroll,
     TransactionItem,
     AnimatedSpinner,
-    Visible,
   },
   props: {
     token: { type: String, default: '' },
-    maxLength: { type: Number, default: null },
-    scrollTopThreshold: { type: Number, default: undefined },
-    showFilters: Boolean,
-    showSearch: Boolean,
     isMultisig: Boolean,
   },
   setup(props, { root }) {
-    const loading = ref(false);
-    const isDestroyed = ref(false);
+    const {
+      activeAccount,
+      accounts,
+    } = useAccounts({ store: root.$store });
+
+    const {
+      activeMultisigAccount,
+    } = useMultisigAccounts({ store: root.$store });
 
     const { isAppActive } = useUi();
-    const { activeAccount } = useAccounts({ store: root.$store });
-    const { activeMultisigAccount } = useMultisigAccounts({ store: root.$store });
 
-    const { pendingMultisigTransaction } = usePendingMultisigTransaction({ store: root.$store });
+    const {
+      viewportElement,
+    } = useViewport();
 
     const {
       searchPhrase,
       displayMode,
       FILTER_MODE,
     } = useTransactionAndTokenFilter();
 
+    const { pendingMultisigTransaction } = usePendingMultisigTransaction({ store: root.$store });
+
+    const loading = ref(false);
+    const isDestroyed = ref(false);
+
     const availableTokens = useState<ITokenList>('fungibleTokens', 'availableTokens');
-    const transactions = useState('transactions');
-    const activeNetwork = useGetter<INetwork>('activeNetwork');
-    const getAccountPendingTransactions = useGetter('getAccountPendingTransactions');
+
+    const transactions = useState<ITransactionsState>('transactions');
     const getTxSymbol = useGetter('getTxSymbol');
+    const getTxDirection = useGetter('getTxDirection');
+    const getDexContracts = useGetter<IDexContracts>('getDexContracts');
+    const getAccountPendingTransactions = useGetter<ITransaction[]>('getAccountPendingTransactions');
 
-    function isFungibleTokenTx(tx: ITx) {
-      return Object.keys(availableTokens.value).includes(tx.contractId);
-    }
+    const fetchTransactions = useDispatch('fetchTransactions');
+
+    const canLoadMore = computed(() => !!transactions.value.nextPageUrl);
 
     const currentAddress = computed(() => props.isMultisig
       ? activeMultisigAccount.value?.gaAccountId
       : activeAccount.value.address);
 
-    const filteredTransactions = computed(
-      () => {
-        const transactionListLocal = [
-          ...getAccountPendingTransactions.value,
-          ...transactions.value.loaded,
-        ];
-
-        if (props.isMultisig && pendingMultisigTransaction.value?.tx) {
-          transactionListLocal.push(pendingMultisigTransaction.value);
-        }
+    const loadedTransactionList = computed((): ICommonTransaction[] => [
+      ...getAccountPendingTransactions.value,
+      ...transactions.value.loaded,
+      ...((props.isMultisig && pendingMultisigTransaction.value?.tx)
+        ? [pendingMultisigTransaction.value!]
```

Comment:
do you need that exclamation mark at the end if you check pendingMultisigTransaction presence in a line above